### ISSUE TITLE
Nested CDATA with tags causes unexpected error 

### ISSUE THREAD

**Issue by markcellus** [NaturalIntelligence/fast-xml-parser#537]
* [x] Are you running the latest version?
* [x] Have you included sample input, output, error, and expected output? 
* [x] Have you checked if you are using correct configuration?
* [x] Did you try online [tool](https://naturalintelligence.github.io/fast-xml-parser/)?

### Description
 
Hello, thanks for building this package! I'm getting an unexpected error when attempting to parse XML with nested CDATA. 
### Code
```js
var fastXmlParser = require("fast-xml-parser")
 const parser = new fastXmlParser.XMLParser();
console.log(parser.parse('<description><![CDATA[<pre><code><category><![CDATA[Architecture]]></category></code></pre><a href="https://test.com">link</a>]]></description>'))
```

### Output

```
 TypeError: Cannot read properties of undefined (reading 'addChild')
```

### expected data

```js
{description: "<pre><code><category><![CDATA[Architecture]]></category></code></pre><a href=\"https://test.com\">link</a>"}
```

**Would you like to work on this issue?**

- [ ] Yes
- [ ] No

[Bookmark](https://github.com/NaturalIntelligence/fast-xml-parser/stargazers) this repository for further updates.

**Comment by amitguptagwl** [comment]
It look like you have nested CDATA which is not allowed by the library.

**Comment by markcellus** [comment]
Right, but shouldn't it just escape the nested CDATA? Or, at the very least, show an error that is more helpful?

**Comment by amitguptagwl** [comment]
have you checked with validator?

**Comment by markcellus** [comment]
Ran it through the validator (output is below). I know that the package considers the XML invalid due to the nested CDATA. But what is the recommended approach of escaping the nested `CDATA`? If you can let me know what that is, I'll just do that and close this issue. Thanks!

-----

```js
var {XMLValidator} = require("fast-xml-parser")
 const result = XMLValidator.validate('<description><![CDATA[<pre><code><category><![CDATA[Architecture]]></category></code></pre><a href="https://test.com">link</a>]]></description>');
console.log(result)
```
Which gives me

```js
{
  err: { 
    code: 'InvalidTag';
    msg: "Expected closing tag 'description' (opened in line 1, col 1) instead of closing tag 'category'.",
    line: 1,
    col: 68 
  };
};
```

**Comment by amitguptagwl** [comment]
currently no.
